modular_skyrat/modules/modular_implants/code/nifsofts/station_pass.dm

```diff
@@ -0,0 +1,191 @@
+	proximity_datum = new(linked_mob, 1)
+	proximity_datum.parent_nifsoft = src
+
+/datum/nifsoft/station_pass/New()
+	qdel(proximity_datum)
+	proximity_datum = null
+
+	return ..()
+
+///Adds a message to the message_list based off the name and message
+/datum/nifsoft/station_pass/proc/add_message(sender_identifier, recieved_name, recieved_message)
+	if(!recieved_message)
+		return FALSE
+	if(!name)
+		name = "Unknown User"
+	else
+		for(var/message in message_list)
+			if(message["identifier"] == sender_identifier)
+				message["sender_name"] = recieved_name
+				message["message"] = recieved_message
+				message["timestamp"] = station_time_timestamp()
+				return TRUE
+
+	message_list.Insert(1, list(list(identifier = sender_identifier, sender_name = recieved_name, message = recieved_message, timestamp = station_time_timestamp())))
+	return TRUE
+
+///Removes a message from the message_list based on the message_to_remove
+/datum/nifsoft/station_pass/proc/remove_message(message_to_remove)
+	var/removed_message = locate(message_to_remove) in message_list
+	if(!removed_message)
+		return FALSE
+
+	message_list -= list(removed_message)
+	return TRUE
+
+/datum/nifsoft/station_pass/activate()
+	. = ..()
+	ui_interact(linked_mob)
+
+
+/datum/nifsoft/station_pass/load_persistence_data()
+	. = ..()
+	var/datum/modular_persistence/persistence = .
+	if(!persistence)
+		return FALSE
+
+	transmitted_name = persistence.station_pass_nifsoft_name
+	transmitted_message = persistence.station_pass_nifsoft_message
+
+/datum/nifsoft/station_pass/save_persistence_data(datum/modular_persistence/persistence)
+	. = ..()
+	if(!.)
+		return FALSE
+
+	persistence.station_pass_nifsoft_message = transmitted_message
+	persistence.station_pass_nifsoft_name = transmitted_name
+
+///The proximty_monitor datum used by the station_pass NIFSoft
+/datum/proximity_monitor/advanced/station_pass
+	///What NIFSoft is this currently attached to?
+	var/datum/weakref/parent_nifsoft
+
+/datum/proximity_monitor/advanced/station_pass/on_entered(turf/source, atom/movable/entered)
+	. = ..()
+	if(host == entered)
+		return FALSE
+
+	var/datum/nifsoft/station_pass/recieving_nifsoft = parent_nifsoft
+	if(!recieving_nifsoft || (!recieving_nifsoft.transmitting_data && !recieving_nifsoft.recieving_data))
+		return FALSE
+
+	var/mob/living/carbon/human/entered_human = entered
+	if(!entered)
+		return FALSE
+
+	var/datum/nifsoft/station_pass/sending_nifsoft = entered_human.find_nifsoft(/datum/nifsoft/station_pass)
+	if(!sending_nifsoft)
+		return FALSE
+
+	if(recieving_nifsoft.recieving_data && sending_nifsoft.transmitting_data)
+		recieving_nifsoft.add_message(sending_nifsoft.transmitted_identifier, sending_nifsoft.transmitted_name, sending_nifsoft.transmitted_message)
+	if(sending_nifsoft.recieving_data && recieving_nifsoft.transmitting_data)
+		sending_nifsoft.add_message(recieving_nifsoft.transmitted_identifier, recieving_nifsoft.transmitted_name, recieving_nifsoft.transmitted_message)
```

Comment:
This feels really bizarre, and should probably be in its own proc, so you just call it by inverting `sending_nifsoft` and `receiving_nifsoft`.

Also it's `receiving` not `recieving`.